090-support-dns-config.md

```diff
@@ -0,0 +1,72 @@
+For example a dev and stage environment/namespace/project hosted in the same Kubernetes/Openshift cluster might need to target seperate target clusters for similar subdomains but are not able to override the nameserver used as this is controlled centrally for both namespaces/projects. 
+By supporting the customization of nameserver configuration for Strimzi resources on the level of the resource this will allow the targeting of specific nameservers. 
+This supports scenarios where resources in different namespaces/projects—while sharing the source cluster network—can still target matching target environments on separate, segregated networks.
+
+## Proposal
+
+By adding `dnsPolicy` and `dnsConfig` properties to Strimzi's common `PodTemplate` in the Strimzi API, users can control domain name resolution for Strimzi resources.
+
+- `dnsPolicy` can specify resolution policies like `ClusterFirst`, `ClusterFirstWithHostNet`, `Default`, and `None`.
+- `dnsConfig` can allow users to define nameservers, search domains, and resolution options.
+
+This provides more control over the `/etc/resolv.conf` configuration for Strimzi resources, allowing domain name resolution to be customized at the resource level.
+Strimzi's CRDs generated from from PodTemplate will thereby allow for user input for name resolution customization.
+By appending these dnsConfig and dnsPolicy properties to WorkloadUtils in the Strimzi cluster operator for PodBuilder and PodTemplateSpecBuilder so that these will propagate to Pods controlled by the operator, the propagation from Strimzi CRD to the Pod resource will be accounted for.
+Provisioned Pods will thereby utilize name resolution configuration based on the user input if specified and fall back to defaults, similar to the current situation, in case no dnsConfig or dnsPolicy is specified by the user. 
+
+An example configuration could look as follows:
+
+```yaml
+kind: KafkaMirrorMaker2
+spec:
+  template:
+    pod:
+      dnsPolicy: "None"
+      dnsConfig:
+        nameservers:
+          - 192.0.2.1 
+        searches:
+          - ns1.svc.cluster-domain.example
+          - my.dns.search.suffix
+        options:
+          - name: ndots
+            value: "2"
+          - name: edns0y:
+```
+When the Pod above is created, the container gets the following contents in its /etc/resolv.conf file:
+
+```
+nameserver 192.0.2.1
+search ns1.svc.cluster-domain.example my.dns.search.suffix
+options ndots:2 edns0
+```
+
+## Affected/not affected projects
+
+Affected:
+- io.strimzi.api for the PodTemplate.
+- io.strimzi.cluster-operator for resources that utilize PodTemplate: kafka, kafkabridge, kafkaconnect, kafkamirrormaker2, kafkanodepool.
+
+## Compatibility
+
+Non-specified dnsConfig and dnsPolicy will not propagate these properties to the Pod resources and thereby comply to the current situation. 
+
+## Rejected alternatives
+
+-
```

Comment:
You can simply remove this if there are no rejected alternatives.

Reply:
Addressed in [c478556](https://github.com/strimzi/proposals/pull/142/commits/c478556265a8dfb09b008544cf91e9ed07523df0)